core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -1125,6 +1125,78 @@ public boolean hasNext() {
     return Streams.stream(rollingIterator(n));
   }
 
+  /**
+   * Transposes data in the table, switching rows for columns.
+   * For example, a table like this.
+   *  label  |  value1  |  value2  |
+   * -------------------------------
+   *   row1  |       1  |       2  |
+   *   row2  |     1.1  |     2.1  |
+   *   row3  |     1.2  |     2.2  |
+   *
+   *  Is transposed into the following
+   *  label   |  row1  |  row2  |  row3  |
+   * -------------------------------------
+   *  value1  |     1  |   1.1  |   1.2  |
+   *  value2  |     2  |   2.1  |   2.2  |
+   *
+   * @return The transposed table
+   */
+  public Table transpose() {
+    Table transposed = Table.create(this.name);
+    Column firstColumn = this.column(0);
+
+    ColumnType[] types = this.columnTypes();
+    //If all columns are of the same type
+    long distinctColumnTypesCount = Arrays.stream(types).skip(1).distinct().count();
+
+    if (distinctColumnTypesCount == 1) {
+      ColumnType resultType = types[1];
+
+      StringColumn labelColumn = StringColumn.create(this.column(0).name());
+      transposed.addColumns(labelColumn);
+      for (String label : firstColumn.asStringColumn()) {
+        transposed.addColumns(resultType.create(label));
+      }
+
+      for (int i = 1; i < this.columnCount(); i++) {
+        Column columnToTranspose = this.column(i);
+        labelColumn.append(columnToTranspose.name());
+      }
+
+      for (int i = 1; i < transposed.columnCount(); i++) {
+        Column column = transposed.column(i);
+        int row = i - 1;
+        for (int col = 1; col < this.columnCount(); col++) {
+          //Avoid boxing for primitives
+          if(ColumnType.DOUBLE == column.type()) {
+            ((DoubleColumn)column).append(this.doubleColumn(col).getDouble(row));
+          }
+          else if(ColumnType.FLOAT == column.type()) {
+            ((FloatColumn)column).append(this.floatColumn(col).getFloat(row));
+          }
+          else if(ColumnType.INTEGER == column.type()) {
```

Comment:
There's some formatting here that doesn't match the rest of the code base. If you compile with maven on the command line then it should auto-format it for you

Reply:
Have reformatted